app/routes/exercises.rb

```diff
@@ -66,6 +66,28 @@ class Exercises < Core
         redirect '/dashboard'
       end
 
+      post '/exercises/:key/request-for-help' do |key|
+        exercise = UserExercise.find_by_key(key)
+        unless current_user.owns?(exercise)
+          flash[:notice] = "Only the author may request help."
+          redirect '/'
+        end
+        exercise.request_help!
+        flash[:success] = "You have requested help on this submission."
+        redirect "/submissions/%s" % exercise.submissions.last.key
+      end
+
+      delete '/exercises/:key/request-for-help' do |key|
+        exercise = UserExercise.find_by_key(key)
+        unless current_user.owns?(exercise)
+          flash[:notice] = "Only the author may cancel a request for help."
+          redirect '/'
+        end
```

Comment:
Code coverage indicates that this branch is not tested.
https://coveralls.io/builds/7232014/source?filename=app%2Froutes%2Fexercises.rb#L82